RecoTracker/MkFitCore/src/CandCloner.cc

```diff
@@ -0,0 +1,169 @@
+                         << "    "
+                         << "original pt=" << ccand[hitsForSeed[ih].trkIdx].pT() << " "
+                         << "nTotalHits=" << ccand[hitsForSeed[ih].trkIdx].nTotalHits() << " "
+                         << "nFoundHits=" << ccand[hitsForSeed[ih].trkIdx].nFoundHits() << " "
+                         << "chi2=" << ccand[hitsForSeed[ih].trkIdx].chi2());
+      }
+#endif
+
+      if (!hitsForSeed.empty()) {
+        //sort the new hits
+        std::sort(hitsForSeed.begin(), hitsForSeed.end(), sortCandListByScore);
+
+        int num_hits = std::min((int)hitsForSeed.size(), mp_iteration_params->maxCandsPerSeed);
+
+        // This is from buffer, we know it was cleared after last usage.
+        std::vector<TrackCand> &cv = t_cands_for_next_lay[is - is_beg];
+
+        int n_pushed = 0;
+
+        for (int ih = 0; ih < num_hits; ih++) {
+          const IdxChi2List &h2a = hitsForSeed[ih];
+
+          TrackCand tc(ccand[h2a.trkIdx]);
+          tc.addHitIdx(h2a.hitIdx, m_layer, h2a.chi2_hit);
+          tc.setScore(h2a.score);
+
+          if (h2a.hitIdx == -2) {
+            if (h2a.score > ccand.m_best_short_cand.score()) {
+              ccand.m_best_short_cand = tc;
+            }
+            continue;
+          }
+
+          // Could also skip storing of cands with last -3 hit.
+
+          // Squeeze in extra tracks that are better than current one.
+          while (extra_i != extra_e && sortByScoreTrackCand(*extra_i, tc) &&
+                 n_pushed < mp_iteration_params->maxCandsPerSeed) {
+            cv.emplace_back(*extra_i);
+            ++n_pushed;
+            ++extra_i;
+          }
+
+          if (n_pushed >= mp_iteration_params->maxCandsPerSeed)
+            break;
+
+          // set the overlap if we have a true hit and pT > pTCutOverlap
+          HitMatch *hm;
+          if (tc.pT() > mp_iteration_params->pTCutOverlap && h2a.hitIdx >= 0 &&
+              (hm = ccand[h2a.trkIdx].findOverlap(h2a.hitIdx, h2a.module))) {
+            tc.addHitIdx(hm->m_hit_idx, m_layer, hm->m_chi2);
+            tc.incOverlapCount();
+
+            // --- ROOT text tree dump of all used overlaps
+            // static bool first = true;
+            // if (first)
+            // {
+            //   // ./mkFit ... | perl -ne 'if (/^ZZZ_EXTRA/) { s/^ZZZ_EXTRA //og; print; }' > extra.rtt
+            //   printf("ZZZ_EXTRA label/I:can_idx/I:layer/I:pt/F:eta/F:phi/F:"
+            //          "chi2_orig/F:chi2/F:chi2_extra/F:module/I:module_extra/I\n");
+            //   first = false;
+            // }
+            // label/I:can_idx/I:layer/I:pt/F:eta/F:phi/F:chi2_orig/F:chi2/F:chi2_extra/F:module/I:module_extra/I
+            // printf("ZZZ_EXTRA %d %d %d %f %f %f %f %f %f %u %u\n",
+            //        tc.label(), h2a.trkIdx, m_layer, tc.pT(), tc.posEta(), tc.posPhi(),
+            //        ccand[h2a.trkIdx].chi2(), h2a.chi2, hm->m_chi2, h2a.module, hm->m_module_id);
```

Comment:
is all this still needed?

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/842cdaf057f74ecf568f12bb1a4cb96e19c5d22c